src/cloudinventario/cloudinventario.py

```diff
@@ -1,122 +1,150 @@
+        mod_instance = mod.setup(collector, config, defaults, options or {})
+
+        # XXX: init for resource collectors (I don't like it)
+        mod_instance._init(
+            collector_pkg=mod_pkg,
+            resources=config.get('collect', [])
+        )
+        return mod_instance
+
+    def login(self, collector, options=None):
+        # workaround for buggy libs
+        wd = os.getcwd()
+        os.chdir("/tmp")
+        try:
+            instance = self.loadCollector(collector, options)
+            instance.login()
+            instance.logout()
+            print(f"+ {collector}: OK")
+            logging.info(f"{collector}: OK")
+            return 0
+        except Exception as e:
+            print(f"+ {collector}: FAILED\nException: {repr(e)}")
+            logging.info("", exc_info=True)
+            logging.info(f"{collector}: FAILED, Exception: {repr(e)}")
+            return 1
+        finally:
+            os.chdir(wd)
+
+    def collect(self, collector, options=None):
+        # workaround for buggy libs
+        wd = os.getcwd()
+        os.chdir("/tmp")
+        inventory = None
+
+        options['metrics']['cloudinventario_source'].inc() if 'metrics' in options else None
+        options['metrics']['cloudinventario_entries_collected'].labels(source=collector).inc() if 'metrics' in options else None
+        try:
+            runtime_start = time.time()
+            instance = self.loadCollector(collector, options)
+            instance.login()
+            inventory = instance.fetch()
+            instance.logout()
+            runtime = time.time() - runtime_start
+
+            options['metrics']['cloudinventario_cpu_usage'].labels(source=collector).set(psutil.cpu_percent(round(runtime))) if 'metrics' in options else None
+            options['metrics']['cloudinventario_mem_usage'].labels(source=collector).set(psutil.virtual_memory()[2]) if 'metrics' in options else None
+            options['metrics']['cloudinventario_runtime'].labels(source=collector).set(runtime) if 'metrics' in options else None
+            options['metrics']['cloudinventario_success'].labels(source=collector).inc() if 'metrics' in options else None
+        except Exception as e:
+          # Find stage from trackback
+            tb = str(traceback.format_exc()).split('\n')
+            for  index, line in enumerate(tb):
+              if 'in collect' in line:
+                stage = re.search(r'instance\.(.*)\(', tb[index+1])
+                stage = stage.group(1) if stage else None 
+                break
+            options['metrics']['cloudinventario_error'].labels(source=collector, stage=stage).inc() if 'metrics' in options else None
+            
+            logging.error(
+                "Exception while processing collector={}".format(collector)) 
+            raise
+        finally:
+            options['pushadd_prometheus'](options['registry']) if 'pushadd_prometheus' in options else None
```

Comment:
Use function (see above)

Reply:
Create function pushMetrics in commit '74c1c26', which perform the test and pushes data into the registry.